src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerMetrics.java

```diff
@@ -156,76 +125,39 @@ public synchronized JsonObject toJson() {
-    if (calls == 0) {
-      json.put("totalSuccessPercentage", 0);
-      json.put("totalErrorPercentage", 0);
-    } else {
-      json.put("totalSuccessPercentage", ((double) success / calls) * 100);
-      json.put("totalErrorPercentage", ((double) (failures + exceptions + timeout) / calls) * 100);
-    }
-
-    addLatency(json, statistics, "total");
+    addSummary(json, rollingWindow.totalSummary(), "total");
 
     // Window metrics
     evictOutdatedOperations();
-    int rollingException = 0;
-    int rollingFailure = 0;
-    int rollingSuccess = 0;
-    int rollingTimeout = 0;
-    int rollingFallbackSuccess = 0;
-    int rollingFallbackFailure = 0;
-    int rollingShortCircuited = 0;
-    Histogram rollingStatistic = new Histogram(3);
-    for (Operation op : window) {
-      rollingStatistic.recordValue(op.durationInMs());
-      if (op.complete) {
-        rollingSuccess = rollingSuccess + 1;
-      } else if (op.failed) {
-        rollingFailure = rollingFailure + 1;
-      } else if (op.exception) {
-        rollingException = rollingException + 1;
-      } else if (op.timeout) {
-        rollingTimeout = rollingTimeout + 1;
-      }
+    addSummary(json, rollingWindow.windowSummary(), "rolling");
 
-      if (op.fallbackSucceed) {
-        rollingFallbackSuccess++;
-      } else if (op.fallbackFailed) {
-        rollingFallbackFailure++;
-      }
+    return json;
+  }
 
-      if (op.shortCircuited) {
-        rollingShortCircuited++;
-      }
-    }
+  private void addSummary(JsonObject json, RollingWindow.Summary summary, String prefix) {
+    long calls = summary.count();
+    int errorCount = summary.failures + summary.exceptions + summary.timeouts;
+
+    json.put(prefix + "OperationCount", calls - summary.shortCircuited);
+    json.put(prefix + "ErrorCount", errorCount);
+    json.put(prefix + "SuccessCount", summary.successes);
+    json.put(prefix + "TimeoutCount", summary.timeouts);
+    json.put(prefix + "ExceptionCount", summary.exceptions);
+    json.put(prefix + "FailureCount", summary.failures);
 
-    json.put("rollingOperationCount", window.size() - rollingShortCircuited);
-    json.put("rollingErrorCount", rollingException + rollingFailure + rollingTimeout);
-    json.put("rollingSuccessCount", rollingSuccess);
-    json.put("rollingTimeoutCount", rollingTimeout);
-    json.put("rollingExceptionCount", rollingException);
-    json.put("rollingFailureCount", rollingFailure);
-    if (window.size() == 0) {
-      json.put("rollingSuccessPercentage", 0);
-      json.put("rollingErrorPercentage", 0);
+    if (calls == 0) {
+      json.put(prefix + "SuccessPercentage", 0);
+      json.put(prefix + "ErrorPercentage", 0);
     } else {
-      json.put("rollingSuccessPercentage", ((double) rollingSuccess / window.size()) * 100);
-      json.put("rollingErrorPercentage",
-        ((double) (rollingException + rollingFailure + rollingTimeout + rollingShortCircuited) / window.size()) * 100);
```

Comment:
This `shortCircuits` were only seen as errors for for the `rollingErrorPercentage`, not for the `totalErrorPercentage`.

Removed it for consistency.  But do not mind to regard them as errors.